app/src/main/scala/org/alephium/explorer/persistence/dao/BlockDao.scala

```diff
@@ -217,5 +218,11 @@ object BlockDao {
         )
       }
     }
+
+    def updateInputs(inputs: Seq[InputEntity]): Future[Int] = {
+      run(
+        DBIOAction.sequence(inputs.map(insertTxPerAddressFromInput))
+      ).map(_.sum)
+    }
```

Comment:
Would there a need for these inserts to be within a transaction? My guess is maybe not because in case of partial success the failed inserts would get retried, is that correct?